src/QueuingEvented.ts

```diff
@@ -9,49 +9,38 @@ import Evented, { isGlobMatch, EventObject } from './Evented';
  *
  * @property maxEvents  The number of events to queue before old events are discarded. If zero (default), an unlimited number of events is queued.
  */
-export default class QueuingEvented extends Evented {
+class QueuingEvented<M extends {} = {}> extends Evented<M> {
 	private _queue: Map<string | symbol, EventObject[]>;
-	private _originalOn: (...args: any[]) => Handle;
 
 	maxEvents = 0;
 
 	constructor() {
 		super();
 
 		this._queue = new Map<string, EventObject[]>();
-		this._originalOn = this.on;
-		this.on = function (this: QueuingEvented, ...args: any[]): Handle {
-			let handle = this._originalOn(...args);
-
-			this.listenersMap.forEach((method, listenerType) => {
-				this._queue.forEach((events, queuedType) => {
-					if (isGlobMatch(listenerType, queuedType)) {
-						events.forEach((event) => this.emit(event));
-						this._queue.delete(queuedType);
-					}
-				});
-			});
-
-			return handle;
-		};
 	}
+}
 
-	emit<E extends EventObject>(event: E): void {
-		super.emit(event);
+(function (proto) {
```

Comment:
How come this is needed?

Reply:
Yes, couldn't something like this be accomplished with the mixin pattern?

Reply:
I'll look into the mixin pattern

Reply:
I think this is left-over from when the signature of `emit` and `on` were much more complex than they are now and I was trying to avoid repeating the signatures in `QueingEvented`. I've fixed these up locally to be methods on `QueingEvented`.